docarray/array/storage/weaviate/find.py

```diff
@@ -27,26 +29,49 @@
 
 
 class FindMixin:
-    def _find_similar_vectors(self, query: 'WeaviateArrayType', limit=10):
+    def _find_similar_vectors(
+        self, query: 'WeaviateArrayType', limit=10, filter: Optional[Dict] = None
+    ):
         query = to_numpy_array(query)
         is_all_zero = np.all(query == 0)
         if is_all_zero:
             query = query + EPSILON
 
         query_dict = {'vector': query}
-        results = (
-            self._client.query.get(
-                self._class_name,
-                ['_serialized', '_additional {certainty}', '_additional {id}'],
+        if filter:
+            results = (
+                self._client.query.get(
+                    self._class_name,
+                    ['_serialized', '_additional {certainty}', '_additional {id}'],
+                )
+                .with_where(filter)
+                .with_limit(limit)
+                .with_near_vector(query_dict)
+                .do()
+            )
+        else:
+            results = (
+                self._client.query.get(
+                    self._class_name,
+                    ['_serialized', '_additional {certainty}', '_additional {id}'],
+                )
+                .with_limit(limit)
+                .with_near_vector(query_dict)
+                .do()
             )
```

Comment:
The `if 'filter'`/`else` here can be done in a more readable way:
```python
        query_builder = (
            self._client.query
            .get(self._class_name, '_serialized')
            .with_additional(['id', 'certainty'])
            .with_limit(limit)
            .with_near_vector(query_dict)

        )
        if filter:
            query_builder = query_builder.with_where(filter)
        results = query_builder.do()
```

Reply:
@StefanBogdan does it always perform pre-filtering in this case, even though the `with_where` statement comes after `with_near_vector` ?

Reply:
Yes, the order does not matter.

Reply:
good catch, updated